runtime/sema/simple_type.go

```diff
@@ -168,3 +168,6 @@ func (t *SimpleType) CompositeKind() common.CompositeKind {
 		return common.CompositeKindStructure
 	}
 }
+
+func (t *SimpleType) CheckInstantiated(pos ast.HasPosition, memoryGauge common.MemoryGauge, report func(err error)) {
+}
```

Comment:
Here and in the rest of the PR:

Remove unused receiver variable and replace unused parameters with `_`. Also, add an explicit `// NO-OP` comment to indicate the empty body is intentional

```suggestion
func (*SimpleType) CheckInstantiated(_ ast.HasPosition, _ common.MemoryGauge, _ func(error)) {
	// NO-OP
}
```

```suggestion
func (t *SimpleType) CheckInstantiated(pos ast.HasPosition, memoryGauge common.MemoryGauge, report func(err error)) {
}
```

Reply:
Done.